### ISSUE TITLE
TypeError: rawIds[i].reduce is not a function

### ISSUE THREAD

**Issue by simllll** [Automattic/mongoose#6460]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
since mongoose 5.1.0 (upgraded from 5.0.4) I get the following error:
TypeError: rawIds[i].reduce is not a function
**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

**Please mention your node.js, mongoose and MongoDB version.**

**Comment by simllll** [comment]
So far what I can add to this issue:
rawIds[i]
i has value 1, and the value of rawIds[1] is an object. existingVal is not an instanceof Map, but it is also an object.

the problematic defintion seems to be somewhere around here:
```
ModelA {
somestructure: {
_ref: [{type: Schema.Types.ObjectId, ref: 'ModelB'}]
}
}
```

maybe because the definition is an array? I will try to find some time on Monday to lock it down

Regards
Simon

**Comment by simllll** [comment]
Okay, somehow the deeppopulate plugin causes this.

works:
```
Model.find({_parent: {$exists: true}})
                .populate({path : '_parent', populate : {path : '_parent'}})
                .exec();
```

Does not work: 
`Model.find({_parent: {$exists: true}}).deepPopulate('_parent._parent').lean().exec();`

So I guess I have to get rid of this plugin, as this feature is included in native mongoose now anyways.

**Comment by vkarpov15** [comment]
@simllll I tried reproducing this issue, to no avail. Can you modify the below script to reproduce your issue?

```javascript
const assert = require('assert');
const mongoose = require('mongoose');
mongoose.set('debug', true);

const GITHUB_ISSUE = `gh6460`;
const connectionString = `mongodb://localhost:27017/${ GITHUB_ISSUE }`;
const { Schema } = mongoose;

run().then(() => console.log('done')).catch(error => console.error(error.stack));

async function run() {
  await mongoose.connect(connectionString);
  await mongoose.connection.dropDatabase();

  mongoose.plugin(require('mongoose-deep-populate')(mongoose));

  const userSchema = new Schema({
    _parent: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User'
    }
  });

  const User = mongoose.model('User', userSchema);

  const doc = new User({});
  doc._parent = doc._id;
  await doc.save();

  await User.find({_parent: {$exists: true}}).deepPopulate('_parent._parent').lean().exec();
}
```

**Comment by simllll** [comment]
Seems to be related with the data of the table. Just tested it with latest mongoose and the same data table, issue still exists.
I can send you the export of it in private if you want to reproduce it.

Regards
Simon

**Comment by vkarpov15** [comment]
@simllll yes please. Shoot me an email, val@karpov.io, we can figure out how to move forward.

**Comment by ycheapopo** [comment]
Still having this issue (5.1.4)

**Comment by vkarpov15** [comment]
Fix will be in 5.1.5
